File: intent-nexus/src/config/urlsCentral.ts
// Centralized frontend URL and variable definitions

export const NETBOX_API_URL = 'http://netbox:8000/api';
export const NETBOX_API_TOKEN = 'your-netbox-token';

export const API_URLS = {
  devices: `${NETBOX_API_URL}/dcim/devices/`,
  interfaces: `${NETBOX_API_URL}/dcim/interfaces/`,
  connections: `${NETBOX_API_URL}/dcim/cables/`,
  sites: `${NETBOX_API_URL}/dcim/sites/`,
  vlans: `${NETBOX_API_URL}/ipam/vlans/`,
  journalEntries: `${NETBOX_API_URL}/extras/journal-entries/`,
  gitApiUrl: 'https://gitlab.com/api/v4',
  configComparisonBaseUrl: 'http://localhost:8000/api/config-comparison',
  ciscoApiUrl: 'https://your-cisco-instance.com:8080/restconf',
  ciscoUsername: 'YOUR_CISCO_USERNAME',
  ciscoPassword: 'YOUR_CISCO_PASSWORD',
};

export { supabase } from '@/integrations/supabase/client';

// Example usage:
// import { API_URLS } from './urlsCentral';
// fetch(API_URLS.devices, ...)

File: intent-nexus/django_backend/urls_central.py
# Centralized backend URL and variable definitions

NETBOX_API_URL = 'http://netbox:8000/api'
NETBOX_API_TOKEN = 'your-netbox-token'

API_URLS = {
    'devices': f'{NETBOX_API_URL}/dcim/devices/',
    'interfaces': f'{NETBOX_API_URL}/dcim/interfaces/',
    'connections': f'{NETBOX_API_URL}/dcim/cables/',
    # Add other backend service URLs here
}

# Example usage:
# from .urls_central import API_URLS
# requests.get(API_URLS['devices'], headers=...)
